Fix python library lookup for RHSCL python 3.6

ExaneServerTeam opened the pull request:
The current patterns does not match the way python is installed
with RedHat SCL python36 on RHEL7.

libpython3.6m.so (libpython%d.%dm.so) points to the right library

htgoebel commented:
Thansk for teh pull-request. There is a similar one #3536 using a different pattern. Can you explain why your patten is different.

htgoebel commented:
Beside this question: For me to merge this, please reword the **commit messages** to comply to our [Commit Message Rules](https://pyinstaller.readthedocs.io/en/latest/development/commit-messages.html#commit-messages). 

You also need to submit a [**changelog entry**](https://pyinstaller.readthedocs.io/en/latest/development/changelog-entries.html) so our users can learn about your change. (This is a new requirement since last September to speed up releasing.)

ExaneServerTeam commented:
Hello @htgoebel,

Thanks for your feedback on this pull requests.  I went through #3536 and we indeed attend to fix the same issue.
The difference between the two implementation, it that this one uses the symlink instead of the library.
The use of the symlink looks more generic than the library itself as the library contains redhat specifics.

`
lrwxrwxrwx.  1 root root      32 Nov 26 11:18 libpython3.4m.so -> libpython3.4m.so.rh-python34-1.0
-rwxr-xr-x.  1 root root 2594960 Aug  5  2016 libpython3.4m.so.rh-python34-1.0
-rwxr-xr-x.  1 root root    6704 Aug  5  2016 libpython3.so.rh-python34
`

By the way I checked SCL python 3.4, 3.5 & 3.6. The three of them use the same library + symlink scheme.

I also updated the git commit message and added the news file as expected.
Please let me know if the updated message & wording is ok with you.

ExaneServerTeam commented:
Thx htgoebel !
